api-put-object.go

```diff
@@ -200,21 +202,129 @@ func (c Client) putObjectCommon(bucketName, objectName string, reader io.Reader,
 		return c.putObjectMultipart(bucketName, objectName, reader, size, metadata, progress)
 	}
 
-	// If size cannot be found on a stream, it is not possible
-	// to upload using streaming signature, fall back to multipart.
 	if size < 0 {
-		// Set regular signature calculation.
-		c.overrideSignerType = credentials.SignatureV4
-		return c.putObjectMultipart(bucketName, objectName, reader, size, metadata, progress)
+		return c.putObjectMultipartStreamNoLength(bucketName, objectName, reader, size, metadata, progress)
 	}
 
-	// Set streaming signature.
-	c.overrideSignerType = credentials.SignatureV4Streaming
-
 	if size < minPartSize {
 		return c.putObjectNoChecksum(bucketName, objectName, reader, size, metadata, progress)
 	}
 
 	// For all sizes greater than 64MiB do multipart.
 	return c.putObjectMultipartStream(bucketName, objectName, reader, size, metadata, progress)
 }
+
+func (c Client) putObjectMultipartStreamNoLength(bucketName, objectName string, reader io.Reader, size int64,
```

Comment:
Sorry for the delay. I don't understand one thing, why we are passing `size` parameter her though this function name is "putObjectMultipartStreamNoLength" (nolength). Besides, this function seems to be called only when size < 0.